File: YSY/11_29/문자열 여러 번 뒤집기.py
def solution(my_string, queries):
    answer = ''
    
    for s, e in queries:
        my_string = list(my_string[:s]) + list(reversed(list(my_string[s:e+1]))) + list(my_string[e+1:])
    answer = ''.join(my_string)
    
    return answer

File: YSY/12_26/문자열 여러 번 뒤집기.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>

using namespace std;

string solution(string my_string, vector<vector<int>> queries) 
{
    string answer = "";
    for (vector<int> query:queries)
    {
        string s(my_string.begin(),my_string.begin()+query[0]);
        string m(my_string.begin()+query[0],my_string.begin()+query[1]+1);
        string e(my_string.begin()+query[1]+1,my_string.end());
        
        reverse(m.begin(),m.end());
        
        my_string=s+m+e;
    }
    answer=my_string;
    return answer;
}